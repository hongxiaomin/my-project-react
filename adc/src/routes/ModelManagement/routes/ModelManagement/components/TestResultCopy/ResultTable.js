import React from 'react'
import propTypes from 'prop-types'
import { connect } from 'react-redux'
import { isEmpty, remove, last } from 'lodash'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import ReactTable from 'components/ADCTable'
import {
  getTestResultList,
} from '../../modules'

export class ResultTable extends React.Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  componentDidMount() {
    this.props.dispatch(getTestResultList(this.props.id))
  }
  render() {
    if (!isEmpty(this.props.testResultList)) {
      const columns = [
        {
          Header: 'Defect Code',
          id: 'defectCode',
          accessor: d => d.defectCode,
        },
        {
          Header: 'Recall',
          id: 'recall',
          accessor: d => d.recall,
        },
      ]
      const keys = Object.keys(this.props.testResultList[0])
      remove(keys, n => (n === 'defectCode' || n === 'recall'))
      keys.map((item) => {
        return columns.splice(1, 0, { Header: item.toString(), accessor: item.toString() })
      })
      const lastData = last(this.props.testResultList)
      const data = this.props.testResultList.filter((item) => {
        return item.defectCode !== 'Replace Rate'
      })
      return (
        <div>
          <ReactTable
            style={{ margin: '10px 30px' }}
            className="-striped -highligh-t"
            defaultPageSize={data.length}
            showPagination={false}
            data={data}
            columns={columns}
            getTrProps={() => {
              return {
                style: { textAlign: 'center', height: '28px', lineHeight: '14px' },
              }
            }}
          />
          <div style={{ display: 'flex', width: '40%', margin: '30px auto', justifyContent: 'space-between' }}>
            <div><WrapperFormattedMessage id="modelManagement.resultTable.correctRateAvg" /> {lastData.correcRateAvg}</div>
            <div><WrapperFormattedMessage id="modelManagement.resultTable.errorRateAvg" /> {lastData.errorRateAvg}</div>
            <div><WrapperFormattedMessage id="modelManagement.resultTable.replaceRate" /> {lastData.replaceRate}</div>
          </div>
        </div>
      )
    } else {
      return <div />
    }
  }
}

ResultTable.propTypes = {
  testResultList: propTypes.array.isRequired,
  dispatch: propTypes.func.isRequired,
  id: propTypes.number.isRequired,
}
export default connect(
  state => ({
    token: state.auth.token,
    testResultList: state.modelManagement.testResultList,
  }),
)(ResultTable)
