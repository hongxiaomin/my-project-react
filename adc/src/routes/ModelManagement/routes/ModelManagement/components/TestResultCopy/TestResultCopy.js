import React from 'react'
import { Slider, InputNumber } from 'antd'
import { isEmpty } from 'lodash'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import ResultTable from './ResultTable'
import classes from './TestResultCopy.scss'

import {
  updateConfidence,
  getTestResultType,
} from '../../modules'

type Props = {
  id: Number,
  dispatch: Function,
  modelResultType: Array,
};

const marks = { 0: '0%', 20: '20%', 40: '40%', 60: '60%', 80: '80%', 100: '100%' }

export class TestResult extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      count: {},
    }
    this.onChange = this.onChange.bind(this)
    this.update = this.update.bind(this)
  }

  componentDidMount() {
    this.props.dispatch(getTestResultType(this.props.id))
  }

// 手动输入/滑动Slider改变confidence
  onChange(keyIndex) {
    return (value) => {
      this.props.dispatch({
        type: 'UPDATE_MODEL_TEST_RESULT_TYPE',
        data: {
          keyIndex,
          confidence: value,
        },
      })
    }
  }
  // 保存更新的数据
  update() {
    const newData = this.props.modelResultType.map((item) => {
      return { defectId: item.defectId.toString(), value: item.defaultConfidence }
    })
    const tmp = {
      modelId: this.props.id,
      thresholds: newData,
    }
    this.props.dispatch(updateConfidence(tmp))
  }

  props: Props;

  render() {
    return (
      <div className={classes['TestResultCopy-container']}>
        <div className={classes['top-left']}>
          <span className={classes.productGlass}>
            <WrapperFormattedMessage id="modelManagement.TestResultCopy.testingResult" />
          </span>
          <button
            className={`pt-button pt-intent-primary ${classes.button}`}
            style={{ width: '80px', marginLeft: '20px' }}
            type='button'
            onClick={this.update}
          >
            <WrapperFormattedMessage id="Save" />
          </button>
        </div>
        <div className={classes.confidence}>
          <div style={{ borderBottom: '1px solid #ccc', paddingBottom: '20px' }}>
            <ResultTable id={this.props.id} />
            <div className={classes.confidenceHeader}>
              <span style={{ fontSize: '16px' }}><WrapperFormattedMessage id="confidence" /></span>
            </div>
            {
              !isEmpty(this.props.modelResultType) &&
              this.props.modelResultType.map((item, index) => {
                return (
                  <div style={{ display: 'flex', marginTop: '30px' }} key={item.defectId}>
                    <div style={{ display: 'flex', marginLeft: '30px', marginTop: '10px', marginBottom: '10px' }}>
                      <span style={{ marginRight: '30px', paddingTop: '5px' }}>{item.defectCode}</span>
                      <Slider
                        className={classes.slider}
                        min={0}
                        max={100}
                        step={1}
                        marks={marks}
                        onChange={this.onChange(index)}
                        value={item.defaultConfidence}
                      />
                      <InputNumber
                        min={0}
                        max={100}
                        step={1}
                        style={{ height: '30px' }}
                        value={item.defaultConfidence}
                        onChange={this.onChange(index)}
                      />
                    </div>
                    {
                      item.defaultConfidence !== 0 ?
                        <div style={{ display: 'flex', marginLeft: '2%', marginTop: '10px', width: '100%' }}>
                          <div style={{ marginLeft: '6%' }}><WrapperFormattedMessage id="recall" /><span>{item.count[item.defaultConfidence - 1].recall}%</span></div>
                          <div style={{ marginLeft: '6%' }}><WrapperFormattedMessage id="errorRate" /><span>{item.count[item.defaultConfidence - 1].error}%</span></div>
                          <div style={{ marginLeft: '6%' }}><WrapperFormattedMessage id="others" /><span>{item.count[item.defaultConfidence - 1].others}%</span></div>
                        </div> :
                        <div style={{ display: 'flex', marginLeft: '2%', marginTop: '10px', width: '100%' }}>
                          <div style={{ marginLeft: '6%' }}><WrapperFormattedMessage id="recall" /><span>0%</span></div>
                          <div style={{ marginLeft: '6%' }}><WrapperFormattedMessage id="errorRate" /><span>0%</span></div>
                          <div style={{ marginLeft: '6%' }}><WrapperFormattedMessage id="others" /><span>0%</span></div>
                        </div>
                    }
                  </div>
                )
              })
            }
          </div>
        </div>
      </div>
    )
  }
}

export default connect(
  state => ({
    token: state.auth.token,
    modelResultType: state.modelManagement.modelResultType,
  }),
)(TestResult)
