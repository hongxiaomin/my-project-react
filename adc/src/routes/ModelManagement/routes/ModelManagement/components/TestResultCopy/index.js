import TestResultCopy from './TestResultCopy'
export default TestResultCopy
