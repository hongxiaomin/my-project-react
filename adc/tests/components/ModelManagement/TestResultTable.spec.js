import React from 'react'
import { shallow } from 'enzyme'
import { ResultTable } from 'routes/ModelManagement/routes/ModelManagement/components/TestResultCopy/ResultTable'

describe('(Component) ResultTable', () => {
  it('data is empty', () => {
    const testResultList = []
    const dispatch = jest.fn()
    const wrapper = shallow(
      <ResultTable testResultList={testResultList} id={2} dispatch={dispatch} />,
    )
    expect(testResultList).toHaveLength(0)
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally', () => {
    const testResultList = [{
      defectCode: 'TNDAC1',
      others: 91,
      TNDAC0: 1030,
      TNDAC1: 600,
      recall: '50%',
    }]
    const dispatch = jest.fn()
    const wrapper = shallow(
      <ResultTable testResultList={testResultList} id={2} dispatch={dispatch} />,
    )
    expect(testResultList).toHaveLength(1)
    expect(wrapper).toMatchSnapshot()
  })
})
