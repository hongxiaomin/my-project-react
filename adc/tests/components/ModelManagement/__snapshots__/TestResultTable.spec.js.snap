// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ResultTable data is empty 1`] = `<div />`;

exports[`(Component) ResultTable render normally 1`] = `
<div>
  <ReactTable
    AggregatedComponent={[Function]}
    ExpanderComponent={[Function]}
    FilterComponent={[Function]}
    LoadingComponent={[Function]}
    NoDataComponent={[Function]}
    PadRowComponent={[Function]}
    PaginationComponent={[Function]}
    PivotValueComponent={[Function]}
    ResizerComponent={[Function]}
    TableComponent={[Function]}
    TbodyComponent={[Function]}
    TdComponent={[Function]}
    TfootComponent={[Function]}
    ThComponent={[Function]}
    TheadComponent={[Function]}
    TrComponent={[Function]}
    TrGroupComponent={[Function]}
    aggregatedKey="_aggregated"
    className="-striped -highligh-t"
    collapseOnDataChange={true}
    collapseOnPageChange={true}
    collapseOnSortingChange={true}
    column={
      Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      }
    }
    columns={
      Array [
        Object {
          "Header": "Defect Code",
          "accessor": [Function],
          "id": "defectCode",
        },
        Object {
          "Header": "TNDAC1",
          "accessor": "TNDAC1",
        },
        Object {
          "Header": "TNDAC0",
          "accessor": "TNDAC0",
        },
        Object {
          "Header": "others",
          "accessor": "others",
        },
        Object {
          "Header": "Recall",
          "accessor": [Function],
          "id": "recall",
        },
      ]
    }
    data={
      Array [
        Object {
          "TNDAC0": 1030,
          "TNDAC1": 600,
          "defectCode": "TNDAC1",
          "others": 91,
          "recall": "50%",
        },
      ]
    }
    defaultExpanded={Object {}}
    defaultFilterMethod={[Function]}
    defaultFiltered={Array []}
    defaultPageSize={1}
    defaultResized={Array []}
    defaultSortDesc={false}
    defaultSortMethod={[Function]}
    defaultSorted={Array []}
    expanderDefaults={
      Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      }
    }
    filterable={false}
    freezeWhenExpanded={false}
    getLoadingProps={[Function]}
    getNoDataProps={[Function]}
    getPaginationProps={[Function]}
    getProps={[Function]}
    getResizerProps={[Function]}
    getTableProps={[Function]}
    getTbodyProps={[Function]}
    getTdProps={[Function]}
    getTfootProps={[Function]}
    getTfootTdProps={[Function]}
    getTfootTrProps={[Function]}
    getTheadFilterProps={[Function]}
    getTheadFilterThProps={[Function]}
    getTheadFilterTrProps={[Function]}
    getTheadGroupProps={[Function]}
    getTheadGroupThProps={[Function]}
    getTheadGroupTrProps={[Function]}
    getTheadProps={[Function]}
    getTheadThProps={[Function]}
    getTheadTrProps={[Function]}
    getTrGroupProps={[Function]}
    getTrProps={[Function]}
    groupedByPivotKey="_groupedByPivot"
    indexKey="_index"
    loading={false}
    loadingText="Loading..."
    nestingLevelKey="_nestingLevel"
    nextText="Next"
    noDataText="No rows found"
    ofText="of"
    onFetchData={[Function]}
    originalKey="_original"
    pageSizeOptions={
      Array [
        5,
        10,
        20,
        25,
        50,
        100,
      ]
    }
    pageText="Page"
    pivotDefaults={Object {}}
    pivotIDKey="_pivotID"
    pivotValKey="_pivotVal"
    previousText="Previous"
    resizable={true}
    rowsText="rows"
    showPageJump={true}
    showPageSizeOptions={true}
    showPagination={false}
    showPaginationBottom={true}
    showPaginationTop={false}
    sortable={true}
    style={
      Object {
        "margin": "10px 30px",
      }
    }
    subRowsKey="_subRows"
  />
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "30px auto",
        "width": "40%",
      }
    }
  >
    <div>
      Correct Rate Avg: 
    </div>
    <div>
      Error Rate Avg: 
    </div>
    <div>
      Replace Rate: 
    </div>
  </div>
</div>
`;
